### PR TITLE
Develop v0.3.0

### PR DESCRIPTION
**Pull request by KCollins**
# Description

Addresses #4 , #6 , #7

Adding documentation, unit tests. 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests have now been added; cf. tests folder.

**Test Configuration**:
* Operating system: Hal
* Version number: Python 3.X
* Any details about your local setup that are relevant

# Checklist:

- [ ] Make sure you are merging into the ``develop`` (not ``main``) branch
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have linted the files updated in this pull request
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings  - 
**New warning: cf #21 **

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] Add a note to ``CHANGELOG.md``, summarizing the changes
- [X] Update zenodo.json file and CITATION.cff file for new code contributors

### CODE REVIEW

File: tests/__init__.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Or, if you do not have the executable installed:
+::
+
+  python -m pytest
+
+To run a specific test, include the filename (with the full path if not in the
+test directory), class, and test name:
+::
+
+  pytest test_utils.py::TestScaleUnits::test_scale_units_angles
```
Comment (aburrell) [inline tests/__init__.txt]:
```suggestion
```

Looks like you're using unittest, not pytest so the calls are different.

File: tests/test_conjcalc.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        expected_lat, expected_lon = 0, 0  # Expect default values
+
+        result_lat, result_lon = conj.findconj(lat, lon, ut, method='invalid_method')
+
+        self.assertEqual(result_lat, expected_lat)
+        self.assertEqual(result_lon, expected_lon)
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (aburrell) [inline tests/test_conjcalc.py]:
```suggestion
```

This is old formatting, you don't need it anymore.

**Review by aburrell — CHANGES_REQUESTED** [review]
You also need to add unit tests to your testing yaml or create a new yaml that runs the unit tests.  Here's an example for a package that uses the same testing framework: https://github.com/aburrell/ocbpy/blob/main/.github/workflows/main.yml

**Comment by aburrell** [comment]
Also, tests would usually be located in `conjugate_map/conjugate_map/tests/` directory, not in the central directory where all your install files are.  Since they're brand new, I'd recommend moving them.

**Comment by KCollins** [comment]
> Also, tests would usually be located in `conjugate_map/conjugate_map/tests/` directory, not in the central directory where all your install files are.  Since they're brand new, I'd recommend moving them.

Can do. Would it make sense to rename
`conjugate_map/conjugate_map` as `conjugate_map/src` as well, or will that break things? (trying to bring things into _SoftwareX_ standards)

**Comment by aburrell** [comment]
No, that's an old standard.  The way you have it set up now will be perfect once the tests are moved.

On another stylistic note, I'd recommend just adding your unit tests to the existing lint tests.  That way the longer unit tests won't run if the linting tests fail first.  Saves a bit of time.

**Comment by KCollins** [comment]
> Also, tests would usually be located in `conjugate_map/conjugate_map/tests/` directory, not in the central directory where all your install files are. Since they're brand new, I'd recommend moving them.

✓ So moved in most recent commit

**Comment by KCollins** [comment]
> No, that's an old standard. The way you have it set up now will be perfect once the tests are moved.
> 
> On another stylistic note, I'd recommend just adding your unit tests to the existing lint tests. That way the longer unit tests won't run if the linting tests fail first. Saves a bit of time.

Attempted to add tests to the pylint workflow and pyproject.toml per the example, but I'm running into this error: 
```
Run coverage run --rcfile=pyproject.toml -m unittest discover
/home/runner/work/_temp/b1722e0a-c156-45e8-9b8b-60d083e58609.sh: line 1: coverage: command not found
Error: Process completed with exit code 127.
```
@aburrell , would you take a look and let me know what I'm missing? :confused:

**Comment by aburrell** [comment]
Ok @KCollins the tests are running now.  They pass for me locally, but fail here.  Why don't you take a look and see if you can figure out what's up with the northern hemisphere geopack test?

**Review by KCollins — COMMENTED** [review]
Approved

**Comment by KCollins** [comment]
> Ok @KCollins the tests are running now. They pass for me locally, but fail here. Why don't you take a look and see if you can figure out what's up with the northern hemisphere geopack test?

Thanks! That's _weird_ - they run locally for me too. A couple of theories:
 - Maybe it's using the default datetime (now) instead of the example time in the test file for some reason
 - Maybe the "1 delta" is more permissive on my computer for some reason?

The value it's getting for the test is very close but still distinctly different: 
```AssertionError: np.float64(66.52335766183909) != 65.38287134065757 within 1 delta (np.float64(1.1404863211815126) difference)```

I will look into it.

**Review by KCollins — COMMENTED** [review]
Approved

**Comment by KCollins** [comment]
> > Ok @KCollins the tests are running now. They pass for me locally, but fail here. Why don't you take a look and see if you can figure out what's up with the northern hemisphere geopack test?
> 
> Thanks! That's _weird_ - they run locally for me too. A couple of theories:
> 
> * Maybe it's using the default datetime (now) instead of the example time in the test file for some reason
> * Maybe the "1 delta" is more permissive on my computer for some reason?
> 
> The value it's getting for the test is very close but still distinctly different: `AssertionError: np.float64(66.52335766183909) != 65.38287134065757 within 1 delta (np.float64(1.1404863211815126) difference)`
> 
> I will look into it.

Figured it out! That test was missing input coordinates. Weird that it ran locally.... Fixed and running on Github now.

**Pull request merged**